staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -280,6 +280,33 @@ func TestExponentialBackoff(t *testing.T) {
 	}
 }
 
+func TestExponentialBackoffWithContext(t *testing.T) {
+	opts := Backoff{Factor: 1.0, Steps: 3}
+
+	// waits up to steps
+	ctx := context.TODO()
+	i := 0
+	err := ExponentialBackoffWithContext(ctx, opts, func(ctx context.Context) (bool, error) {
+		i++
+		return false, nil
+	})
+	if err != ErrWaitTimeout || i != opts.Steps {
```

Comment:
opts.Steps is mutated, no?

Reply:
a copy of opts is passed in... the original is not modified